python/ray/experimental/state.py

```diff
@@ -352,69 +358,105 @@ def log_files(self):
+
         # Parse through event logs to determine task start and end points.
-        for i in range(len(event_log_sets)):
-            event_list = self.redis_client.zrangebyscore(event_log_sets[i],
-                                                         min=start,
-                                                         max=end,
-                                                         start=start,
-                                                         num=num)
-            for event in event_list:
+        for event_log_set in event_log_sets:
+            if start is None and end is None:
+                if fwd:
+                    event_list = self.redis_client.zrange(
+                        event_log_set,
+                        **params)
+                else:
+                    event_list = self.redis_client.zrevrange(
+                        event_log_set,
+                        **params)
+            else:
+                if fwd:
+                    event_list = self.redis_client.zrangebyscore(
+                        event_log_set,
+                        **params
+                        )
+                else:
+                    event_list = self.redis_client.zrevrangebyscore(
+                        event_log_set,
+                        **params
+                     )
+
+            for (event, score) in event_list:
                 event_dict = json.loads(event)
                 task_id = ""
                 for event in event_dict:
                     if "task_id" in event[3]:
                         task_id = event[3]["task_id"]
                 task_info[task_id] = dict()
+                task_info[task_id]["score"] = score
+                # Add task to (min/max) heap by its start point.
+                # if fwd, we want to delete the largest elements, so -score
+                if num_slice is not None:
+                    heapq.heappush(heap, (-score if fwd else score, task_id))
+                    heap_size += 1
+
                 for event in event_dict:
                     if event[1] == "ray:get_task" and event[2] == 1:
                         task_info[task_id]["get_task_start"] = event[0]
-                        # Add task to min heap by its start point.
-                        heapq.heappush(heap,
-                                       (task_info[task_id]["get_task_start"],
-                                        task_id))
-                        heap_size += 1
                     if event[1] == "ray:get_task" and event[2] == 2:
                         task_info[task_id]["get_task_end"] = event[0]
-                    if (event[1] == "ray:import_remote_function" and
-                            event[2] == 1):
-                        task_info[task_id]["import_remote_start"] = event[0]
-                    if (event[1] == "ray:import_remote_function" and
-                            event[2] == 2):
+                    if event[1] == "ray:import_remote_function" and \
```

Comment:
Let's not use `\` for multiline things. Instead, this should be formatted something like

```python
if (event[1] == "ray:import_remote_function"
        and event[2] == 1):
```